File: 第一章/learning_topic/scripts/student_subscriber.py
#!/usr/bin/env python
# coding:utf-8

import rospy
from learning_topic.msg import Student

count = 0

def studentInfoCallback(msg):
    rospy.loginfo("Subscribe Student Info: name:%s age:%d sex:%d height:%d weight:%d ", 
			 msg.name, msg.age, msg.sex, msg.height, msg.weight)
    global count 
    count+=1
    rospy.loginfo("Subscribe times are:[%d]", count)  
def student_subscriber():
	# ROS节点初始化
    rospy.init_node('student_subscriber', anonymous=True)

	# 创建一个Subscriber，订阅名为/person_info的topic，注册回调函数personInfoCallback
    rospy.Subscriber("/student_info", Student, studentInfoCallback)

	# 循环等待回调函数
    rospy.spin()

if __name__ == '__main__':
    student_subscriber()

File: chapter 2/learning_topic/src/student_subscriber.cpp
#include <ros/ros.h>
#include "learning_topic/Student.h"

int count = 0;
void studentInfoCallback(const learning_topic::Student::ConstPtr& msg)
{
    ROS_INFO("Subscribe Student Info: name:%s age:%d sex:%d height:%d weight:%d ", 
			 msg->name.c_str(), msg->age, msg->sex, msg->height, msg->weight);
    ROS_INFO("Subscribe times are:[%d]" ,++::count);
}

int main(int argc, char **argv)
{
    ros::init(argc, argv, "student_subscriber");

    ros::NodeHandle nh;

    ros::Subscriber student_info_sub = nh.subscribe("/student_info", 1000, studentInfoCallback);

    ros::spin();

    return 0;
}